### PR TITLE
Stage binaries to temp location

### PR DESCRIPTION
**Pull request by krishvoor**
This PR aims to stage the required binaries to /tmp
location and not override the system defaults.

Signed-off-by: Krishna Harsha Voora <krvoora@redhat.com>

### CODE REVIEW

**Comment by krishvoor** [comment]
Hi @mukrishn ptal

File: workloads/hyper-scale/common.sh [hunk 1]
```diff
@@ -2,12 +2,13 @@
...
 prep(){
     if [[ -z $(go version) ]]; then
         curl -L https://go.dev/dl/go1.18.2.linux-amd64.tar.gz -o go1.18.2.linux-amd64.tar.gz
-        tar -C /usr/local -xzf go1.18.2.linux-amd64.tar.gz
-        export PATH=$PATH:/usr/local/go/bin
+        tar -C ${TEMP_DIR}/ -xzf go1.18.2.linux-amd64.tar.gz
+        export PATH=${TEMP_DIR}/go/bin/:$PATH
     fi
     if [[ ${HYPERSHIFT_CLI_INSTALL} != "false" ]]; then
         echo "Remove current Hypershift CLI directory.."
```
Comment (mukrishn) [inline workloads/hyper-scale/common.sh:14]: @krishvoor why aren't we doing this for hypershift binaries ?

Comment (krishvoor) [reply workloads/hyper-scale/common.sh:14]: @mukrishn here's the one, where the changes are introduced
https://github.com/cloud-bulldozer/e2e-benchmarking/pull/463/files

**Comment by krishvoor** [comment]
Hi @smalleni @dry923 ptal

File: workloads/hyper-scale/common.sh [hunk 2]
```diff
@@ -2,12 +2,13 @@
...
 source env.sh
+TEMP_DIR=`mktemp -d`
 
 prep(){
     if [[ -z $(go version) ]]; then
         curl -L https://go.dev/dl/go1.18.2.linux-amd64.tar.gz -o go1.18.2.linux-amd64.tar.gz
-        tar -C /usr/local -xzf go1.18.2.linux-amd64.tar.gz
-        export PATH=/usr/local/go/bin:$PATH
+        tar -C ${TEMP_DIR}/ -xzf go1.18.2.linux-amd64.tar.gz
+        export PATH=${TEMP_DIR}/go/bin/:$PATH
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:11]: Don't think we need the trailing / on ${TEMP_DIR}/go/bin/

File: workloads/hyper-scale/common.sh [hunk 3]
```diff
@@ -2,12 +2,13 @@
 set -x
 
 source env.sh
+TEMP_DIR=`mktemp -d`
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:5]: Let's remove the old export TEMP_DIR line (line #15) if we are moving it here

File: workloads/hyper-scale/common.sh [hunk 4]
```diff
@@ -19,14 +20,16 @@ prep(){
...
     fi
     if [[ -z $(rosa version)  ]]; then
-        sudo curl -L $(curl -s https://api.github.com/repos/openshift/rosa/releases/latest | jq -r ".assets[] | select(.name == \"rosa-linux-amd64\") | .browser_download_url") --output /usr/local/bin/rosa
-        sudo curl -L $(curl -s https://api.github.com/repos/openshift-online/ocm-cli/releases/latest | jq -r ".assets[] | select(.name == \"ocm-linux-amd64\") | .browser_download_url") --output /usr/local/bin/ocm
-        sudo chmod +x /usr/local/bin/rosa && chmod +x /usr/local/bin/ocm
+        mkdir -p ${TEMP_DIR}/bin/
+        sudo curl -L $(curl -s https://api.github.com/repos/openshift/rosa/releases/latest | jq -r ".assets[] | select(.name == \"rosa-linux-amd64\") | .browser_download_url") --output ${TEMP_DIR}/bin/rosa
+        sudo curl -L $(curl -s https://api.github.com/repos/openshift-online/ocm-cli/releases/latest | jq -r ".assets[] | select(.name == \"ocm-linux-amd64\") | .browser_download_url") --output ${TEMP_DIR}/bin/ocm
+        sudo chmod +x ${TEMP_DIR}/bin/rosa && chmod +x ${TEMP_DIR}/bin/ocm
+        export PATH=${TEMP_DIR}/bin/:$PATH
```
Comment (dry923) [inline workloads/hyper-scale/common.sh:27]: Training / again. Just for consistency w/ what is normally in $PATH

File: workloads/hyper-scale/common.sh (hunk 2 above)
Comment (krishvoor) [reply workloads/hyper-scale/common.sh]: Done!

File: workloads/hyper-scale/common.sh (hunk 3 above)
Comment (krishvoor) [reply workloads/hyper-scale/common.sh:5]: Done, updated.

File: workloads/hyper-scale/common.sh (hunk 4 above)
Comment (krishvoor) [reply workloads/hyper-scale/common.sh]: Done, updated!

**Review by dry923 — APPROVED** [review]
LGTM!

File: workloads/hyper-scale/common.sh [hunk 5]
```diff
@@ -2,31 +2,33 @@
...
         make build
         export PATH=$TEMP_DIR/hypershift/bin:$PATH
         popd
     fi
     if [[ -z $(rosa version)  ]]; then
-        sudo curl -L $(curl -s https://api.github.com/repos/openshift/rosa/releases/latest | jq -r ".assets[] | select(.name == \"rosa-linux-amd64\") | .browser_download_url") --output /usr/local/bin/rosa
-        sudo curl -L $(curl -s https://api.github.com/repos/openshift-online/ocm-cli/releases/latest | jq -r ".assets[] | select(.name == \"ocm-linux-amd64\") | .browser_download_url") --output /usr/local/bin/ocm
-        sudo chmod +x /usr/local/bin/rosa && chmod +x /usr/local/bin/ocm
+        mkdir -p ${TEMP_DIR}/bin/
+        sudo curl -L $(curl -s https://api.github.com/repos/openshift/rosa/releases/latest | jq -r ".assets[] | select(.name == \"rosa-linux-amd64\") | .browser_download_url") --output ${TEMP_DIR}/bin/rosa
```
Comment (morenod) [inline workloads/hyper-scale/common.sh:23]: As we are using custom folders, we can remove sudo

**Review by morenod — APPROVED** [review]
LGTM

**Review by mukrishn — APPROVED** [review]

**Comment by smalleni** [comment]
@mukrishn @dry923 merge?

**Pull request merged**